azext_iot/iothub/providers/discovery.py

```diff
@@ -30,9 +33,9 @@ def _initialize_client(self):
         if not self.client:
             if getattr(self.cmd, "cli_ctx", None):
                 self.client = iot_hub_service_factory(self.cmd.cli_ctx)
+                self.sub_id = get_subscription_id(self.cmd.cli_ctx)
             else:
                 self.client = self.cmd
-            self.sub_id = self.client.config.subscription_id
```

Comment:
We _may_ hit a regression here since before `sub_id` was extracted when the client was initialized and used to build the `target` dictionary.  Might help determine problems if we change this https://github.com/c-ryan-k/azure-iot-cli-extension/blob/sdk_track_2_support/azext_iot/tests/iothub/test_iothub_discovery_int.py#L102 to assert sub_id is not the "unknown" placeholder value.